src/traces/isosurface/convert.js

```diff
@@ -0,0 +1,205 @@
+
+      singlemesh: true,
+
+      colorscale: 'Portland',
+      capscolorscale: 'Jet'
+    }], {
+      scene: {
+        xaxis: {range: [0, 1]},
+        yaxis: {range: [0, 1]},
+        zaxis: {range: [0, 1]}
+      }
+    })
+
+*/
+'use strict';
+
+var isosurfacePlot = require('gl-isosurface3d');
+
+var simpleMap = require('../../lib').simpleMap;
+var parseColorScale = require('../../lib/gl_format_color').parseColorScale;
+
+function Isosurface(scene, uid) {
+    this.scene = scene;
+    this.uid = uid;
+    this.mesh = null;
+    this.data = null;
+}
+
+var proto = Isosurface.prototype;
+
+proto.handlePick = function(selection) {
+    if(selection.object === this.mesh) {
+        var selectIndex = selection.index = selection.data.index;
+
+        selection.traceCoordinate = [
+            selection.data.position[0],
+            selection.data.position[1],
+            selection.data.position[2],
+            selection.data.intensity
+        ];
+
+        var text = this.data.text;
+        if(Array.isArray(text) && text[selectIndex] !== undefined) {
+            selection.textLabel = text[selectIndex];
+        } else if(text) {
+            selection.textLabel = text;
+        }
+
+        return true;
+    }
+};
+
+var axisName2scaleIndex = {xaxis: 0, yaxis: 1, zaxis: 2};
+
+function getSequence(src) {
+    var xs = [src[0]];
+    for(var i = 1, last = xs[0]; i < src.length; i++) {
+        var p = src[i];
+        if(p >= last) {
+            if(p > last) {
+                xs.push(p);
+            }
+            last = p;
+        } else {
+            break;
+        }
+    }
+    return xs;
+}
+
+function convert(scene, trace) {
+    var sceneLayout = scene.fullSceneLayout;
+    var dataScale = scene.dataScale;
+    var isosurfaceOpts = {};
+
+    function toDataCoords(arr, axisName) {
+        var ax = sceneLayout[axisName];
+        var scale = dataScale[axisName2scaleIndex[axisName]];
+        return simpleMap(arr, function(v) { return ax.d2l(v) * scale; });
+    }
+
+    var xs = getSequence(trace.x);
+    var ys = getSequence(trace.y);
+    var zs = getSequence(trace.z);
+
+    isosurfaceOpts.dimensions = [xs.length, ys.length, zs.length];
+    isosurfaceOpts.meshgrid = [
+        toDataCoords(xs, 'xaxis'),
+        toDataCoords(ys, 'yaxis'),
+        toDataCoords(zs, 'zaxis')
+    ];
+
+    // var bounds = [
+    //    isosurfaceOpts.boundmin || [xs[0], ys[0], zs[0]],
+    //    isosurfaceOpts.boundmax || [xs[xs.length - 1], ys[ys.length - 1], zs[zs.length - 1]]
+    // ];
+
+
+    isosurfaceOpts.values = trace.u;
+
+    isosurfaceOpts.colormap = parseColorScale(trace.colorscale);
+    // isosurfaceOpts.capsColormap = parseColorScale(trace.capscolorscale);
+    isosurfaceOpts.vertexIntensityBounds = [trace.cmin, trace.cmax];
+    isosurfaceOpts.isoBounds = [trace.imin, trace.imax];
+
+    isosurfaceOpts.isoCaps = trace.isocaps;
+    isosurfaceOpts.singleMesh = trace.singlemesh === undefined ? true : trace.singlemesh;
```

Comment:
What does `singlemesh` do? BTW logic like this belongs as `attributes.singlemesh.dlft = true`.